pip/flatpak-pip-generator

```diff
@@ -74,12 +74,45 @@ def get_file_hash(filename: str) -> str:
             sha.update(data)
         return sha.hexdigest()
 
+
+def parse_continuation_lines(fin):
+    for line in fin:
+        line = line.rstrip('\n')
+        while line.endswith('\\'):
+            line = line[:-1] + next(fin).rstrip('\n')
+        yield line
+
+
+def parse_hashes(line):
+    name = line.split(" ")[0]
+    hash_requirements = []
+    while "--hash=" in line:
+        split = line.split("--hash=", 1)
+        hash_algorithm = split[1].split(":", 1)[0]
+        hash_value = split[1].split(":", 1)[1].split(" ", 1)[0]
+        line = line.replace('--hash=' + hash_algorithm + ":" + hash_value, '')
+        hash_requirements.append([hash_algorithm.lower(), hash_value])
+    return name, hash_requirements
+
+
 packages = []
+use_hash = None
 if opts.requirements_file and os.path.exists(opts.requirements_file):
     with open(opts.requirements_file, 'r') as req_file:
-        packages = list(requirements.parse(req_file))
+        reqs = list(parse_continuation_lines(req_file))
```

Comment:
Why `list()`

Reply:
I am not happy with it either. What I do is to iter over `reqs` and change its contents (via `reqs[i]=foo`). Later `list()` is used again for the `packages` iterable so `len(packages) == 1` can be called without consuming the iterable, I haven't tough of a more graceful way of handling it.